### PR TITLE
Laboratory work #3, Andrew Zapekin - 23FPL1

### PR DESCRIPTION
**Pull request by Andrew-cmd216**

### CODE REVIEW

File: lab_2_retrieval_w_bm25/main.py [hunk 1]
```diff
@@ -42,11 +42,17 @@ def remove_stopwords(tokens: list[str], stopwords: list[str]) -> list[str] | Non
 
     In case of corrupt input arguments, None is returned.
     """
+<<<<<<< HEAD
```
Comment (EIzharikov) [inline lab_2_retrieval_w_bm25/main.py:45]: what is it?

File: lab_2_retrieval_w_bm25/main.py [hunk 2]
```diff
@@ -68,6 +74,18 @@ def build_vocabulary(documents: list[list[str]]) -> list[str] | None:
             not all(isinstance(document, list) for document in documents) or \
             not documents:
         return None
+<<<<<<< HEAD
```
Comment (EIzharikov) [inline lab_2_retrieval_w_bm25/main.py:77]: fix all conflicts!!

**Review by EIzharikov — COMMENTED** [review]
Nothing to review :( You should hurry up and start your hard work! This lab work is harder than 1 and 2. Good luck! I believe you!

File: lab_2_retrieval_w_bm25/main.py (hunk 1 above)
Comment (Andrew-cmd216) [reply lab_2_retrieval_w_bm25/main.py]: i don't know how it is happened, but I fixed it

File: lab_2_retrieval_w_bm25/main.py (hunk 2 above)
Comment (Andrew-cmd216) [reply lab_2_retrieval_w_bm25/main.py]: done

File: lab_2_retrieval_w_bm25/main.py [hunk 3]
```diff
@@ -245,8 +245,11 @@ def rank_documents(
 
     result = []
     for i, document in enumerate(indexes):
```
Comment (EIzharikov) [inline lab_2_retrieval_w_bm25/main.py:247]: revert changes with lab_2

File: lab_2_retrieval_w_bm25/start.py [hunk 4]
```diff
@@ -2,13 +2,18 @@
 Laboratory Work #2 starter
 """
 # pylint:disable=too-many-locals, unused-argument, unused-variable, too-many-branches, too-many-statements, duplicate-code
+
+from main import (build_vocabulary, calculate_bm25, calculate_idf, rank_documents, remove_stopwords,
```
Comment (EIzharikov) [inline lab_2_retrieval_w_bm25/start.py:6]: revert changes with lab_2

File: lab_3_ann_retriever/main.py [hunk 5]
```diff
@@ -130,6 +161,13 @@ def _remove_stop_words(self, tokens: list[str]) -> list[str] | None:
 
         In case of corrupt input arguments, None is returned.
         """
+        if (not isinstance(tokens, list) or not all(isinstance(token, str) for token in tokens)
+                or not tokens):
+            return None
+
+        list_without_sw = [token for token in tokens if token not in self.stop_words]
```
Comment (EIzharikov) [inline lab_3_ann_retriever/main.py:168]: return as is

**Review by EIzharikov — COMMENTED** [review]
Nice progress, fix CI and continue your work!

File: lab_3_ann_retriever/main.py (hunk 5 above)
Comment (Andrew-cmd216) [reply lab_3_ann_retriever/main.py:168]: done

File: lab_2_retrieval_w_bm25/main.py (hunk 3 above)
Comment (EIzharikov) [reply lab_2_retrieval_w_bm25/main.py:247]: still-waiting-meme-mens-t-shirt

File: lab_2_retrieval_w_bm25/start.py (hunk 4 above)
Comment (EIzharikov) [reply lab_2_retrieval_w_bm25/start.py:6]: still-waiting-meme-mens-t-shirt

File: lab_3_ann_retriever/main.py [hunk 6]
```diff
@@ -304,6 +415,15 @@ def retrieve_vectorized(self, query_vector: Vector) -> str | None:
 
         In case of corrupt input arguments, None is returned.
         """
+        if not isinstance(query_vector, tuple):
+            return None
+        for vector in self._document_vectors:
+            if len(vector) < len(query_vector):
```
Comment (EIzharikov) [inline lab_3_ann_retriever/main.py:421]: why are you checking like this? what will be if `len(vector > len(query_vector)` ?? nothing?

**Review by EIzharikov — COMMENTED** [review]
Make CI green and fix comments, Good luck for Friday

File: lab_2_retrieval_w_bm25/main.py (hunk 3 above)
Comment (Andrew-cmd216) [reply lab_2_retrieval_w_bm25/main.py:247]: done

File: lab_2_retrieval_w_bm25/start.py (hunk 4 above)
Comment (Andrew-cmd216) [reply lab_2_retrieval_w_bm25/start.py:6]: done

File: lab_3_ann_retriever/main.py (hunk 6 above)
Comment (Andrew-cmd216) [reply lab_3_ann_retriever/main.py]: yeah, I understood and fixed this

**Pull request closed without merging**